src/net.cpp

```diff
@@ -2713,7 +2723,10 @@ void CConnman::StopNodes()
             if (anchors_to_dump.size() > MAX_BLOCK_RELAY_ONLY_ANCHORS) {
                 anchors_to_dump.resize(MAX_BLOCK_RELAY_ONLY_ANCHORS);
             }
-            DumpAnchors(gArgs.GetDataDirNet() / ANCHORS_DATABASE_FILENAME, anchors_to_dump);
+
+            if (tried_connect_anchors) {
```

Comment:
If `anchors_to_dump` is empty, there is nothing to dump.

```suggestion
            if (!anchors_to_dump.empty()) {
```

Reply:
What if my node connected to one node from anchors.dat but a connection to the other one has not been tried yet? In this case, `anchors_to_dump` wouldn't be empty.